arch/arm/src/tiva/common/tiva_can.c

```diff
@@ -0,0 +1,2401 @@
+             canmod->modnum);
+    }
+
+  modifyreg32(TIVA_SYSCON_SRCAN, 0, SYSCON_SRCAN(modnum));
+
+  /* Spin for a few clock cycles */
+
+  for (int i = 0; i < 16; ++i);
+
+  modifyreg32(TIVA_SYSCON_SRCAN, SYSCON_SRCAN(modnum), 0);
+
+  /* Wait for peripheral-ready signal */
+
+  while (!(getreg32(TIVA_SYSCON_PRCAN) & (1 << modnum)));
+}
+
+/****************************************************************************
+ * Name: tivacan_setup
+ *
+ * Description:
+ *   Set up the CAN module. Register the ISR and enable the interrupt in the
+ *   NVIC. Set default bit-timing and set filters to a consistent state. On
+ *   return, the INIT bit is cleared in the CANCTL register, and interrupts
+ *   are enabled on the side of the CAN module. (can_ops_s.txint and
+ *   can_ops_s.rxint are unsuitable because the Tiva CAN modules do not allow
+ *   disabling TX and RX interrupts separately).
+ *
+ *   A pointer to this function is passed to can_register.
+ *
+ * Input parameters:
+ *   dev - An instance of the "upper half" CAN driver structure
+ *
+ * Returned value:
+ *   Zero on success, or a negated errno on failure.
+ ****************************************************************************/
+
+static int tivacan_setup(FAR struct can_dev_s *dev)
+{
+  uint32_t  irq;
+  int       ret;
+  uint32_t  reg;
+  FAR struct tiva_canmod_s    *canmod = dev->cd_priv;
+  char     *kthd_argv[2];
+  kthd_argv[1] = NULL;
+
+  ret = nxsem_init(&canmod->rxsem, 0, 0);
+
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  switch (canmod->modnum)
+    {
+#ifdef CONFIG_TIVA_CAN0
+      case 0:
+        {
+          kthd_argv[0] = "0";
+          ret = kthread_create("Tiva CAN0",
+                               CONFIG_TIVA_CAN0_PRIO,
+                               TIVA_CAN_KTHREAD_STACKSIZE,
+                               tivacan_rxhandler,
+                               kthd_argv);
+        }
+      break;
+#endif
+#ifdef CONFIG_TIVA_CAN1
+      case 1:
+        {
+          kthd_argv[0] = "1";
+          ret = kthread_create("Tiva CAN1",
+                               CONFIG_TIVA_CAN1_PRIO,
+                               TIVA_CAN_KTHREAD_STACKSIZE,
+                               tivacan_rxhandler,
+                               kthd_argv);
+        }
+      break;
+#endif
+      default:
+        return -EINVAL;
+    }
+
+  if (ret < 0)
+    {
+      canerr("tiva_can: failed to create kthread\n");
+      return ret;
+    }
+  else
+    {
+      canmod->kthd_id = ret;
+    }
+
+#ifdef CONFIG_CAN_EXTID
+  struct canioc_extfilter_s default_filter =
+  {
+    .xf_id1   = 0x123,
+    .xf_id2   = 0,
+    .xf_type  = CAN_FILTER_MASK,
+    .xf_prio  = CAN_MSGPRIO_HIGH,
+  };
+#else
+  struct canioc_stdfilter_s default_filter =
+  {
+    .sf_id1   = 0x123,
+    .sf_id2   = 0,
+    .sf_type  = CAN_FILTER_MASK,
+    .sf_prio  = CAN_MSGPRIO_HIGH,
+  };
+#endif
+
+  /* REVISIT: This will not necessarily come out nicely on boards with
+   * different crystals.
+   */
+
```

Comment:
Can this be calculated from SYSCLK_FREQUENCY defined in the board's board.h?

Reply:
See comment: https://github.com/apache/incubator-nuttx/pull/5151#issuecomment-1008236696